File: SharedPage/wwwroot/js/jsecharts.js

export class echartsFunc {
    static liChart = {};
    static ResizeListener = {};
    static init(id) {
        try {
            var ele = document.getElementById(id);
            if (ele) {
                var myChart = echarts.init(ele);
                if (myChart) {
                    echartsFunc.liChart[id] = myChart;
                    console.log('初始化成功', echartsFunc.liChart,id);
                }
                else {
                    console.log('初始化失败', echartsFunc.liChart,id);
                }
            }
        } catch (e) {
            console.log('初始化失败',e);
        }

    }
    static setOption(id, option, notMerge, lazyUpdate) {
        try {
            var chart = echartsFunc.liChart[id];
            if (chart) {
                let opt = eval('(' + option + ')');
                chart.setOption(opt);
              //  chart.setOption(option);
            }
            else {
                console.log('没有这个chart实例', id);
            }
        } catch (e) {
            console.log(e);
        }
    }
    static dispose(id) {
        try {
            var chart = echartsFunc.liChart[id];
            if (chart) {
                if (chart.isDisposed()) {
                    chart.dispose();//测试永远不会走到这里,不知道是哪里执行了销魂
                }
                else {
                    //console.log("已经被销毁了");
                }
                delete echartsFunc.liChart[id];
            }
            else {
                //console.log('已经被销毁了', id);
            }
        } catch (e) {

        }
    }
    static resize(id) {
        try {
            if (echartsFunc.liChart[id]) {
                echartsFunc.liChart[id].resize();
            }
        } catch (e) {
            console.log('resize失败', e);
        }
        
    }
    static addResizeListener(id) {
        echartsFunc.ResizeListener[id] = () => { echartsFunc.liChart[id].resize(); };
        window.addEventListener("resize", echartsFunc.ResizeListener[id]);
     
    }
    static removeResizeListener(id) {
        window.removeEventListener("resize", echartsFunc.ResizeListener[id]);

    }
    static addClassForSelect(id) {
        var ele = document.getElementById(id);
        if (!ele.classList.contains('selected')) {
            ele.classList.add('selected');
        }
    }
    static removeClassForSelect() {
        try {
            for (var id in echartsFunc.liChart) {
                console.log(id);
                var ele = document.getElementById(id);
                if (ele) {
                    if (ele.classList.contains('selected')) {
                        ele.classList.remove('selected');
                    }
                }
            }
        }
        catch (e) {
            console.log(e);
            console.log('cuocuole');
        }


    }
    static getWH(id) {
        var w = document.getElementById(id).clientWidth;
        var h = document.getElementById(id).clientHeight;
        return [w, h]
    }
    static Log(id) {
        console.log(id);
    }

}

File: SharedPage/JsEcharts.cs
using Microsoft.JSInterop;

namespace SharedPage
{
    // This class provides an example of how JavaScript functionality can be wrapped
    // in a .NET class for easy consumption. The associated JavaScript module is
    // loaded on demand when first needed.
    //
    // This class can be registered as scoped DI service and then injected into Blazor
    // components for use.

    public class JsEcharts : IAsyncDisposable
    {
        private readonly Lazy<Task<IJSObjectReference>> moduleTask;

        public JsEcharts(IJSRuntime jsRuntime)
        {
            moduleTask = new(() => jsRuntime.InvokeAsync<IJSObjectReference>(
                "import", "./_content/SharedPage/js/jsecharts.js").AsTask());
        }
        public async ValueTask Init(string id)
        {
            var module = await moduleTask.Value;
            await module.InvokeVoidAsync("echartsFunc.init", id);
        }
        public async ValueTask SetOption(string id,object? option)
        {
            var module = await moduleTask.Value;
            await module.InvokeVoidAsync("echartsFunc.setOption", id,option);
        }
        public async ValueTask dispose(string id)
        {
            var module = await moduleTask.Value;
            await module.InvokeVoidAsync("echartsFunc.dispose", id);
        }
        
        public async ValueTask Resize(string id)
        {
            var module = await moduleTask.Value;
            await module.InvokeVoidAsync("echartsFunc.resize", id);
        }
        public async ValueTask addResizeListener(string id)
        {
            var module = await moduleTask.Value;
            await module.InvokeVoidAsync("echartsFunc.addResizeListener", id);
        }
        public async ValueTask removeResizeListener(string id)
        {
            var module = await moduleTask.Value;
            await module.InvokeVoidAsync("echartsFunc.removeResizeListener", id);
        }
        public async ValueTask addClassForSelect(string id)
        {
            var module = await moduleTask.Value;
            await module.InvokeVoidAsync("echartsFunc.addClassForSelect", id);
        }
        public async ValueTask removeClassForSelect()
        {
            var module = await moduleTask.Value;
            await module.InvokeVoidAsync("echartsFunc.removeClassForSelect");
        }
        public async ValueTask<string> Prompt(string message)
        {
            var module = await moduleTask.Value;
            return await module.InvokeAsync<string>("showPrompt", message);
        }


        public async ValueTask<int[]> getWH(string id)
        {
            var module = await moduleTask.Value;
            return await module.InvokeAsync<int[]>("echartsFunc.getWH", id);
        }
        public async ValueTask Prompt2(string id)
        {
            var module = await moduleTask.Value;
            await module.InvokeVoidAsync($"echarts.init(document.getElementById({id}))");
        }
        public async ValueTask Log(object id)
        {
            var module = await moduleTask.Value;
            await module.InvokeVoidAsync("echartsFunc.Log",id);
        }
        public async ValueTask DisposeAsync()
        {
            if (moduleTask.IsValueCreated)
            {
                var module = await moduleTask.Value;
                await module.DisposeAsync();
            }
        }
    }
}
